libs/base/System/Concurrency.idr

```diff
@@ -187,6 +183,10 @@ data Channel : Type -> Type where [external]
 prim__makeChannel : PrimIO (Channel a)
 %foreign "scheme:blodwen-channel-get"
 prim__channelGet : Channel a -> PrimIO a
+%foreign "scheme,chez:blodwen-channel-get-non-blocking"
+prim__channelGetNonBlocking : Channel a -> PrimIO (Maybe a)
+%foreign "scheme,chez:blodwen-channel-get-with-timeout"
+prim__channelGetWithTimeout : Channel a -> Nat -> PrimIO (Maybe a)
```

Comment:
Last I checked `Nat` was not a blessed primitive type for passing via the FFI (https://idris2.readthedocs.io/en/latest/ffi/ffi.html#primitive-ffi-types). What that means in practice is that if it works then it "just happens to work" even though `Nat` may in practice be implemented in a way that is compatible with `Int` for any given backend.

Someone could correct me here, but best practice would be to use `Nat` in your exported code but cast to `Int` for your FFI call.

Reply:
@mattpolzin 

Awesome catch here, I've addressed this via eddaf30b1cb60f30b1bc0367927391df71416ef6

Reply:
@mattpolzin 

It looks like all tests passed but one:

https://github.com/idris-lang/Idris2/actions/runs/12659854176/job/35280037440?pr=3435

It looks like it timed out, but not sure if that is supposed to happen?

Reply:
It shouldn't time out but that's not unheard of so I've just kicked tests off again.